File: CarRental/Dtos/CarDto.cs
using CarRental.Helpers;
using CarRental.Models;
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CarRental.Dtos
{
    public class CarDto
    {

        public int Id { get; set; }
        public string RegistrationNumber { get; set; }
        public int Year { get; set; }
        public string Make { get; set; }
        public string Model { get; set; }
        public long Mileage { get; set; }
        public bool isAvailable { get; set; }
        public bool isActive { get; set; }
        public CarClassificationDto CarClassification { get; set; }
        public string Image { get; set; }
       public IList<CarAvailbilityTimeDto> CarAvailability { get; set; }
        

        public int PassengerCount { get; set; }
        public float CostPerHour { get; set; }
        public float CostPerDay { get; set; }
        public float LateFeePerHour { get; set; }

    }
}

File: CarRental/ClientApp/src/app/classes/CarDto.ts
import { CarClassificationDto } from "./CarClassificationDto";
import { CarAvailbilityTimeDto } from "./CarAvailbilityTimeDto";

 

    // $Classes/Enums/Interfaces(filter)[template][separator]
    // filter (optional): Matches the name or full name of the current item. * = match any, wrap in [] to match attributes or prefix with : to match interfaces or base classes.
    // template: The template to repeat for each matched item
    // separator (optional): A separator template that is placed between all templates e.g. $Properties[public $name: $Type][, ]

    // More info: http://frhagn.github.io/Typewriter/

    
    export class CarDto {
        
        // Id
        public Id: number = 0;
        // RegistrationNumber
        public RegistrationNumber: string = null;
        // Year
        public Year: number = 0;
        // Make
        public Make: string = null;
        // Model
        public Model: string = null;
        // Mileage
        public Mileage: number = 0;
        // isAvailable
        public isAvailable: boolean = false;
        // isActive
        public isActive: boolean = false;
        // CarClassification
        public CarClassification: CarClassificationDto = null;
        // Image
        public Image: string = null;
        // CarAvailability
        public CarAvailability: CarAvailbilityTimeDto[] = [];
    }

